addons/mail/static/src/web/activity/activity_service.js

```diff
@@ -53,17 +55,43 @@ export class ActivityService {
      * @param {import("./activity_model").Data} data
      * @returns {import("./activity_model").Activity}
      */
-    insert(data) {
+    insert(data, fromBroadcastChannel = false) {
         const activity = this.store.activities[data.id] ?? new Activity(this.store, data.id);
         if (data.request_partner_id) {
             data.request_partner_id = data.request_partner_id[0];
         }
         assignDefined(activity, data);
+        if (!fromBroadcastChannel) {
+            this.broadcastChannel.postMessage({
+                type: "insert",
+                payload: this._serialize(activity),
+            });
+        }
         return activity;
     }
 
-    delete(activity) {
+    delete(activity, fromBroadcastChannel = false) {
         delete this.store.activities[activity.id];
+        if (!fromBroadcastChannel) {
+            this.broadcastChannel.postMessage({ type: "delete", payload: { id: activity.id } });
+        }
+    }
+
+    _onBroadcastChannelMessage({ data }) {
+        switch (data.type) {
+            case "insert":
+                this.insert(data.payload, true);
+                break;
+            case "delete":
+                this.delete(data.payload, true);
```

Comment:
Not great to have positional argument as boolean.
Prefer naming them, e.g. `{ initiator: true }`